src/library_webgpu.js

```diff
@@ -624,9 +658,75 @@ var LibraryWebGPU = {
   wgpuDeviceCreateBindGroupLayout: function(deviceId, descriptor) {
     {{{ gpu.makeCheckDescriptor('descriptor') }}}
 
-    function makeEntry(entryPtr) {
+    function makeBufferEntry(entryPtr) {
       {{{ gpu.makeCheck('entryPtr') }}}
 
+      var type = WebGPU.BufferBindingType[
+        {{{ gpu.makeGetU32('entryPtr', C_STRUCTS.WGPUBufferBindingLayout.type) }}}]
+
+      if (type === undefined)
+        return undefined;
+
+      return {
+        "type": type,
```

Comment:
nit: Doing the array lookup isn't really necessary to check this. As a tiny optimization you could check whether the int value of the type is 0:

```suggestion
      var typeInt =
        {{{ gpu.makeGetU32('entryPtr', C_STRUCTS.WGPUBufferBindingLayout.type) }}};
      if (!typeInt) return undefined;

      return {
        "type": WebGPU.BufferBindingType[typeInt],
```

Reply:
Done, and in all the other places as well